lib/util/url.ts

```diff
@@ -15,7 +15,7 @@ export function ensurePathPrefix(url: string, prefix: string): string {
 }
 
 export function ensureTrailingSlash(url: string): string {
-  return url.replace(/\/?$/, '/'); // TODO #12070 #12071 add tests for this one
+  return url.replace(/\/?$/, '/'); // TODO #12070 #12071 adds / in the front when used with re2
```

Comment:
Super weird

Reply:
Yes I had the same thought, then I tried it by in a different environment and got the same result.
Is it possible to check it on your side, so that we can be clear it is not due to some error on my system.

Reply:
The same error occurs in some other cases too when using similar regex. I think an issue on the node-re2 repo will bring more clarity on this one as I couldn't find any information in the docs, except for this [statement](Can't resolve './myjourney/path' in 'C:\Users\LENOVO\Desktop\elearning-learner-2\src\components\myjourney\').

Reply:
```
❯ node
Welcome to Node.js v16.13.0.
Type ".help" for more information.
> const RE2 = require('re2');
undefined
> console.log("hello".replace(RE2(/\/?$/), '/'));
/hello
undefined
>  
```

Reply:
This type of very simple difference concerns me

Reply:
This works:

```
> console.log("hello".replace(/([^/])?$/, '$1/'));
hello/
undefined
> console.log("hello".replace(RE2(/([^/])?$/), '$1/'));
hello/
```